resin/auth.py

```diff
@@ -105,9 +105,13 @@ def is_logged_in(self):
 
         """
 
-        return self.base_request.request(
-            '/whoami', 'GET', endpoint=self.settings.get('api_endpoint')
-        )
+        try:
+            self.base_request.request(
+                '/whoami', 'GET', endpoint=self.settings.get('api_endpoint')
+            )
+            return True
+        except:
```

Comment:
I understand that :)

However, if the token has expired, which exception is thrown? Is it Python's general ``Exception``, an exception from the ``requests`` module, an exception subclass of yours?

All I'm saying is that you should catch these specific exceptions there, not irrelevant exceptiosn, like one resulting from bad network -- these must be able to propagate.